### PR TITLE
Add "Intended" Door of Time option

### PR DESCRIPTION
**Pull request by Kewlan**
When set to "Intended", the Door of Time will only open when the Song of Time is played on the Ocarina of Time while having all three spiritual stones.

~If it matters, the "correct song played" tune still plays when playing the Song of Time without the necessary requirements.~

### CODE REVIEW

**Review by cjohnson57 — APPROVED** [review]
Looks good to me

**Comment by Kewlan** [comment]
New commits from Freddy:
- Playing SoT to open the DoT with the "intended" setting will play the error sfx when the door can't be opened
- The spiritual stones on the ToT pedestal will only appear if the DoT is open, even for adult Link

**Review by gamestabled — COMMENTED** [review]
Everything here looks good to me, but I have one more request: Can we edit the child altar text depending on this setting? Right now it always says `"Ye who may become a Hero...&Stand with the Ocarina and&play the Song of Time."`

File: source/hints.cpp [hunk 1]
```diff
@@ -501,6 +501,26 @@ static Text BuildDungeonRewardText(ItemID itemID, const ItemKey itemKey) {
   return Text()+ITEM_OBTAINED(itemID)+"#"+GetHintRegion(Location(location)->GetParentRegionKey())->GetHint().GetText()+"#...^";
 }
 
+static Text BuildDoorOfTimeText() {
+  std::string itemObtained;
+  Text doorOfTimeText;
+
+  if (OpenDoorOfTime.Is(OPENDOOROFTIME_OPEN)) {
+    itemObtained = ITEM_OBTAINED(ITEM_SWORD_MASTER);
```
Comment (Kewlan) [inline source/hints.cpp:509]: I don't know if there's a way to remove the spacing for the image; When I kept `itemObtained` empty it used the previous image, and when I set it to `ITEM_NONE` it kept the spacing with no image. So I set it to the Master Sword as it is the path to the future.

Comment (gamestabled) [reply source/hints.cpp:509]: what if in that case you just return `Text() + doorOfTimeText` without the itemObtained?

Comment (Kewlan) [reply source/hints.cpp:509]: Then it still reuses the previous image like so:
image
image

Comment (gamestabled) [reply source/hints.cpp:509]: gotcha. then master sword image sounds good for now. If there is ever some time of "shuffle master sword" in the future, we can come back to it

**Pull request merged**